web/config.py

```diff
@@ -912,6 +912,14 @@ def get_variables_from_module(module_name):
             file_dir_name = os.path.basename(locals().get(setting))
             locals().update({setting: os.path.join(data_dir, file_dir_name)})
 
+# To use psycopg3 driver, need to specify +psycopg in conn URI
+if 'CONFIG_DATABASE_URI' in user_config_settings:
+    CONFIG_DATABASE_URI = user_config_settings['CONFIG_DATABASE_URI']
+    if CONFIG_DATABASE_URI.startswith('postgresql:'):
+        user_config_settings['CONFIG_DATABASE_URI'] = \
+            'postgresql+psycopg:{0}'.format(
+                CONFIG_DATABASE_URI[CONFIG_DATABASE_URI.find(':') + 1:])
```

Comment:
We are removing psycopg2, so this logic is not required I guess. Instead, update the comments for "CONFIG_DATABASE_URI" in the same file. If in the future the string is changed then won't need to change the logic.

Reply:
Even if we use only psycopg3, the `+psycopg` is required by SQLAlchemy to use psycopg3.

Reply:
Logic like that shouldn't be in config.py. That file is supposed to have an absolute minimum of code in it - ideally, nothing more than platform specific conditionals to set default values accordingly ("if Windows then foo else bar" type stuff).

Reply:
@dpage config is imported at multiple places directly. Although CONFIG_DATABASE_URI is used at only one or two places, but if it is required anywhere then it has to be modified like above before use.
I can move it somewhere else if you say so.

Reply:
Then maybe we need a wrapper of some kind that we import instead, then can load and manipulate settings as we need. Ultimately, config.py is supposed to be a configuration file, not application code. It just happens to use Python syntax...

Reply:
@dpage Or may be we create a new config_defaults.py which will have all the configurations of config.py and config.py (the wrapper) can have all the manipulations? This will make it work with minimum code changes.

Reply:
I am still inclined toward removing this logic completely and updating the uses like "postgresql+psycopg://username:password@host:port/database" in the comment of the config.py which is already there. Just need to add "+psycopg" in each example.

Reply:
> I am still inclined toward removing this logic completely and updating the uses like "postgresql+psycopg://username:password@host:port/database" in the comment of the config.py which is already there. Just need to add "+psycopg" in each example.

@akshay-joshi This will immediately break working pgAdmin for external DB users upon upgrade. Also, users do not need to know `+psycopg` syntax, that is specific to SQLAlchemy. pgAdmin should take care of it.

Reply:
@dpage Creating evaluated_config.py as a wrapper.

Reply:
@dpage If we want to keep config.py clean then we can move the manipulation code to some other file and just call it inside config.py?

Reply:
pushed the change - https://github.com/pgadmin-org/pgadmin4/pull/5915/files#diff-07a6e9f47f63fa72551fe22f3b87ad2935a0806843fd56351b3e5a6c7bc57c6dL862